### DISCUSSION TITLE
[General]: Is `embla-carousel` headless?

### DISCUSSION THREAD

**Discussion by zaaakher** [davidjerleke/embla-carousel#693] [General]
It seems that the current cool thing in the frontend dev community is headless UI (not the library, but the concept). Similar to [Tanstack's headless table](https://tanstack.com/table/v8) and I've seen headless pagination and there's also [paco's command menu](https://github.com/pacocoursey/cmdk) and many others. And people seem to be using "unstyled" and "headless" interchangeably.

`embla-carousel` to me seems to fall within that category. But I could be mistaken. Nevertheless, if it is, then I'd recommend we include that in the tagline or the docs or the READMEs, which could theoretically increase adaptation. 

I'm just thinking out loud here. @davidjerleke let me know what you think 💭

**Comment by davidjerleke** [comment]
Hi @zaaakher!

> `embla-carousel` to me seems to fall within that category. 

Yes, Embla Carousel is as headless as a carousel script can get. Because if you think about it, it lets you use your own HTML and CSS. The only styles it applies are `translate3d` to create the scroll effect, and `translate3d` to some slides in order to create the loop effect when loop is enabled. And these two style manipulations are necessary. 

I think this sums it up from the [get started page](https://www.embla-carousel.com/get-started/):

*Embla Carousel is a library agnostic, dependency free and lightweight carousel library. It aims to solve the hardest technical challenges with building carousels, and the rest is up to the user utilizing its highly extensible [API](https://www.embla-carousel.com/api/) and [plugins](https://www.embla-carousel.com/plugins/). Embla Carousel works in all modern browsers.*

>Nevertheless, if it is, then I'd recommend we include that in the tagline or the docs or the READMEs, which could theoretically increase adaptation.

That sounds good. I’m open to suggestions that would improve the tagline 🙂.

Best,
David

**Reply by zaaakher** [reply]
> Embla Carousel is a library agnostic, dependency free and lightweight carousel library. It aims to solve the hardest technical challenges with building carousels, and the rest is up to the user utilizing its highly extensible [API](https://www.embla-carousel.com/api/) and [plugins](https://www.embla-carousel.com/plugins/). Embla Carousel works in all modern browsers.

Well put, it sums it up well indeed 👍

Although maybe I was using the wrong term with "tagline" I suppose I meant something more like a prefix to the word "carousel" every time it's mentioned. (or at least in the proper context)

For example something like this 👇

![image](https://github.com/davidjerleke/embla-carousel/assets/46135573/b34b1697-4351-43af-9b07-16b87878e93c)

and in the repo bio:

![image](https://github.com/davidjerleke/embla-carousel/assets/46135573/0acb3a5f-76e7-4f34-826c-94057281a7a9)

**Reply by davidjerleke** [reply]
@zaaakher thank you for your ideas! I think updating the repo bio makes sense and the same sentence is also in all Embla `package.json` files. I’ll just have to double check the definition of **headless** before proceeding.

But I don’t want to replace Embla with headless because that’s the carousel brand 🙂.

Thanks as always!

Best,
David

**Reply by zaaakher** [reply]
Hey @davidjerleke sorry for the late reply

> But I don’t want to replace Embla with headless

Yes of course that should remain. I was just suggesting to add the word "headless" in the hero section of the website and in the bio. 

Adding "headless" is merely a marketing thing and some might even categorize it as a buzz word at this point. Feel free to discard my suggestion anytime you want. I'm never offended by that 😁
